provider/aws/aws_test.go

```diff
@@ -1247,6 +1247,19 @@ func TestAWSsubmitChangesRetryOnError(t *testing.T) {
 	require.True(t, containsRecordWithDNSName(records, "fail__edns_housekeeping.zone-1.ext-dns-test-2.teapot.zalan.do"))
 }
 
+func TestAWSnewChangesWithNilEndpoint(t *testing.T) {
+	provider, _ := newAWSProvider(t, endpoint.NewDomainFilter([]string{"ext-dns-test-2.teapot.zalan.do."}), provider.NewZoneIDFilter([]string{}), provider.NewZoneTypeFilter(""), defaultEvaluateTargetHealth, false, nil)
+
+	endpoints := []*endpoint.Endpoint{
+		nil,
+	}
+
+	cs := make(Route53Changes, 0, len(endpoints))
+	cs = append(cs, provider.newChanges(route53.ChangeActionCreate, endpoints)...)
+
+	require.Equal(t, 0, len(cs))
```

Comment:
Given that `endpoints` is a singleton slice containing `nil`, there should be no changes.